Packs/AWS-SNS/Integrations/AWSSNS/AWSSNS.py

```diff
@@ -0,0 +1,407 @@
+
+    except Exception as e:
+        return raise_error(e)
+
+
+def delete_topic(args, client):
+    try:
+        response = client.delete_topic(TopicArn=args.get('topicArn'))
+        if response['ResponseMetadata']['HTTPStatusCode'] == 200:
+            return 'The Topic has been deleted'
+
+    except Exception as e:
+        return raise_error(e)
+
+
+def test_function(aws_client):
+    try:
+        client = aws_client.aws_session(service='sns')
+        response = client.list_topics()
+        if response['ResponseMetadata']['HTTPStatusCode'] == 200:
+            return "ok"
+    except Exception as e:
+        return raise_error(e)
+
+
+def main():
+
+    params = demisto.params()
+    aws_default_region = params.get('defaultRegion')
+    aws_role_arn = params.get('roleArn')
+    aws_role_session_name = params.get('roleSessionName')
+    aws_role_session_duration = params.get('sessionDuration')
+    aws_role_policy = None
+    aws_access_key_id = params.get('access_key')
+    aws_secret_access_key = params.get('secret_key')
+    verify_certificate = not params.get('insecure', True)
+    timeout = params.get('timeout')
+    retries = params.get('retries') or 5
+
+    commands = {
+        'aws-sns-create-subscription': create_subscription,
+        'aws-sns-list-topics': list_topics,
+        'aws-sns-list-subscriptions-by-topic': list_subscriptions_by_topic,
+        'aws-sns-send-message': send_message,
+        'aws-sns-create-topic': create_topic,
+        'aws-sns-delete-topic': delete_topic
+    }
+
+    try:
+        validate_params(aws_default_region, aws_role_arn, aws_role_session_name, aws_access_key_id,
+                        aws_secret_access_key)
+        aws_client = AWSClient(aws_default_region, aws_role_arn, aws_role_session_name, aws_role_session_duration,
+                               aws_role_policy, aws_access_key_id, aws_secret_access_key, verify_certificate, timeout,
+                               retries)
+        command = demisto.command()
+        args = demisto.args()
+        demisto.debug('Command being called is {}'.format(command))
+        if command == 'test-module':
+            return_results(test_function(aws_client))
+        elif command in commands:
+            client = aws_client.aws_session(
+                service='sns',
+                region=args.get('region'),
+                role_arn=args.get('roleArn'),
+                role_session_name=args.get('roleSessionName'),
+                role_session_duration=args.get('roleSessionDuration'))
+            return_results(commands[command](args, client))
+        else:
+            raise NotImplementedError('{} is not an existing AWS-SNS command'.format(command))
+
+    except Exception as e:
+        return_error("Failed to execute {} command.\nError:\n{}".format(demisto.command(), str(e)))
+
+
+### GENERATED CODE ###: from AWSApiModule import *  # noqa: E402
+# This code was inserted in place of an API module.
```

Comment:
please remove all the lines from here until line 402 
it's the API modules code 
and it should automatically updated when the integration will upload to XSOAR
by using

Reply:
@ilappe Done